polaris-core/src/main/java/org/apache/polaris/core/persistence/AtomicOperationMetaStoreManager.java

```diff
@@ -893,6 +893,50 @@ private void revokeGrantRecord(
         : new PrincipalSecretsResult(secrets);
   }
 
+  @Override
+  public @Nonnull PrincipalSecretsResult resetPrincipalSecrets(
```

Comment:
This method looks generic enough... Would it be possible to implement it in `BaseMetaStoreManager`?